packages/contracts/contracts/voting/admin-address/AdminAddressSetup.sol

```diff
@@ -0,0 +1,120 @@
+    address private constant NO_ORACLE = address(0);
+
+    /// @notice Thrown if admin address is zero.
+    /// @param admin The admin address.
+    error AdminAddressInvalid(address admin);
+
+    /// @notice The contract constructor, that deployes the `AdminAddress` plugin logic contract.
+    constructor() {
+        adminAddressBase = address(new AdminAddress());
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallationDataABI() external pure returns (string memory) {
+        return "(address admin)";
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallation(address _dao, bytes memory _data)
+        external
+        returns (
+            address plugin,
+            address[] memory helpers,
+            PermissionLib.ItemMultiTarget[] memory permissions
+        )
+    {
+        IDAO dao = IDAO(_dao);
+
+        // Decode `_data` to extract the params needed for deploying and initializing `AdminAddress` plugin.
+        address admin = abi.decode(_data, (address));
+
+        if (admin == address(0)) {
+            revert AdminAddressInvalid({admin: admin});
+        }
+
+        // Clone plugin contract.
+        plugin = adminAddressBase.clone();
+
+        // Initialize cloned plugin contract.
+        AdminAddress(plugin).initialize(dao);
+
+        // Prepare helpers
+        helpers = new address[](0);
+
+        // Prepare permissions
+        permissions = new PermissionLib.ItemMultiTarget[](2);
+
+        // Grant `ADMIN_EXECUTE_PERMISSION` of the Plugin to the admin.
+        permissions[0] = PermissionLib.ItemMultiTarget(
+            PermissionLib.Operation.Grant,
+            plugin,
+            admin,
+            NO_ORACLE,
+            AdminAddress(plugin).ADMIN_EXECUTE_PERMISSION_ID()
+        );
+
+        // Grant `EXECUTE_PERMISSION` of the DAO to the plugin.
+        permissions[1] = PermissionLib.ItemMultiTarget(
+            PermissionLib.Operation.Grant,
+            _dao,
+            plugin,
+            NO_ORACLE,
+            DAO(payable(_dao)).EXECUTE_PERMISSION_ID()
+        );
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareUninstallationDataABI() external pure returns (string memory) {
+        return "";
+    }
+
+    /// @inheritdoc IPluginSetup
+    /// @dev Currently there are not a relaiable mean to revoke plugin's permission such as `ADMIN_EXECUTE_PERMISSION_ID`
+    /// that have been granted to addresses during the life cycle of the plugin.
+    /// or the ones that have been granted are not revoked yet already,
+    /// therefore, only `EXECUTE_PERMISSION_ID` is revoked for this uninstallation.
+    function prepareUninstallation(
+        address _dao,
+        address _plugin,
+        address[] calldata,
+        bytes calldata
+    ) external view returns (PermissionLib.ItemMultiTarget[] memory permissions) {
+        // Prepare permissions
+        permissions = new PermissionLib.ItemMultiTarget[](1);
+
```

Comment:
Shouldn't we revoke the admin's permission on the plugin as well?

Reply:
+1

Reply:
and where do you have admin address ? This is one of the topics I need to bring to the attention when we are all on the call.